public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -11,89 +21,310 @@ export default class ScheduledLaunch extends React.Component {
+              disabled={!video || videoEditOpen}
+            >
+              Remove indefinite embargo
+            </button>
+          </li>
+        )}
+      </ul>
+    );
+  };
+
+  renderAlert = invalidDateError =>
+    invalidDateError && (
+      <span className="topbar__alert">{invalidDateError}</span>
+    );
+
+  renderDatePicker = (showDatePicker, propertyName, selectedScheduleDate, selectedEmbargoDate) => (
+    showDatePicker &&
+    <DatePicker
+      editable={true}
+      onUpdateField={date => this.setDate(date, propertyName)}
+      fieldValue={
+        propertyName === 'selectedScheduleDate'
+          ? selectedScheduleDate
+          : selectedEmbargoDate
+      }
+    />
+  )
+
+  renderSaveButton = (propertyName, selectedScheduleDate, selectedEmbargoDate, invalidDateError) => (
+    <button
+      className="button__secondary--confirm"
+      onClick={() => this.saveDate(propertyName)}
+      disabled={
+        invalidDateError ||
+        ((propertyName === 'selectedScheduleDate' && !selectedScheduleDate) ||
+          (propertyName === 'selectedEmbargoDate' && !selectedEmbargoDate))
+      }
+    >
+      Save
+    </button>
+  )
+
+  renderRemoveButton = propertyName => (
+    <button
+      className="button__secondary--remove"
+      onClick={() => this.removeDate(propertyName)}
+    >
+      Remove
+    </button>
+  )
+
+  renderCancelButton = () => (
+    <button
+      className="button__secondary--cancel"
+      onClick={() =>
+        this.setState({
+          showDatePicker: false
+        })
+      }
+    >
+      Cancel
+    </button>
+  )
+
+  renderSchedulerButton = (showScheduleOptions) => (
+    <button
+      className="btn btn--list"
+      onClick={() =>
+        this.setState({
+          showScheduleOptions: !showScheduleOptions,
+          propertyName: null
+        })
+      }
+    >
+      <Icon icon="access_time" />
+    </button>
+  )
 
   render() {
-    const { video, video: { contentChangeDetails }, videoEditOpen } = this.props;
-    const { selectedDate } = this.state;
+    const {
+      video,
+      video: { contentChangeDetails },
+      videoEditOpen,
+      hasPublishedVideoUsages
+    } = this.props;
+    const {
+      selectedScheduleDate,
+      selectedEmbargoDate,
+      showScheduleOptions,
+      propertyName
+    } = this.state;
     const showDatePicker = this.state.showDatePicker && !videoEditOpen;
-    const isFutureDate = selectedDate && moment(selectedDate).isAfter(moment());
-    const scheduledLaunch = contentChangeDetails && contentChangeDetails.scheduledLaunch && contentChangeDetails.scheduledLaunch.date;
+    const invalidDateError = this.state.invalidDateError;
+    const scheduledLaunch =
+      contentChangeDetails &&
+      contentChangeDetails.scheduledLaunch &&
+      contentChangeDetails.scheduledLaunch.date;
+    const embargo =
+      contentChangeDetails &&
+      contentChangeDetails.embargo &&
+      contentChangeDetails.embargo.date;
```

Comment:
Did we not already access the scheduled launch of the video on lines 50-58?

Reply:
I need to access them in the `render()` function, but I'd like to avoid having to call `validateDate()` with two extra arguments, as it is called by both `setDate()` and `onSelectOption()`, so then I would have to pass the extra arguments to these two functions as well.